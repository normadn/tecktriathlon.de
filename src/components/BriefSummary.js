import React from "react";
import { css } from "glamor";

import ContentWrapper from "./ContentWrapper";

import cubes1webp from "../assets/img/icons/cubes-1.webp";
import cubes1png from "../assets/img/icons/cubes-1.png";

const caretCSS = (backgroundColor) =>
  css({
    backgroundColor,
    width: "26px",
    height: "4px"
  });

const BriefSummary = () => (
  <div
    css={{
      position: "relative",
      width: "100%",
      background: "#f8f8f8",
      paddingTop: "80px",
      paddingBottom: "80px"
    }}
  >
    <ContentWrapper>
      <div
        css={{
          background: "#ffffff",
          display: "grid",
          gridTemplateColumns: "auto auto",
          padding: "80px 55px 80px 55px",
          margin: "0 auto 0 auto",
          borderRadius: "8px",
          boxShadow: "4px 9px 40px 2px rgba(7, 16, 75, 0.1)",
          border: "solid 1px #e8e8e8",
          "@media(max-width: 750px)": {
            padding: "32px 24px 32px 24px"
          },
          "@media(max-width: 1115px)": {
            display: "block"
          }
        }}
      >
        <div
          css={{
            paddingRight: "43px",
            "@media (max-width: 920px)": { paddingRight: "0px" }
          }}
        >
          <h1 css={{ paddingTop: "2px" }}>Dream It. Build It.</h1>
          <div
            css={{
              display: "grid",
              gridTemplateColumns: "auto auto 1fr",
              gridColumnGap: "4px",
              padding: "13px 0px 27px 0px"
            }}
          >
            <div {...caretCSS("#00205b")} />
            <div {...caretCSS("#c81c2e")} />
            <div {...caretCSS("#fedb01")} />
          </div>
          <p css={{ lineHeight: "24px", fontColor: "#000000" }}>
            QHacks is back in 2019 for another year of collaboration, ideation,
            and creativity! On February 1st - 3rd, more than 500 students from
            all over North America will come together to turn their ideas into
            reality. Attend workshops, play with hardware, take part in hacker
            challenges, and meet some new friends. Out-of-the-box thinkers,
            creatives, and innovators from all faculties and skill levels are
            encouraged to attend. We can&#39;t wait to see you there!
          </p>
        </div>
        <div
          css={{
            margin: "auto auto auto auto",
            "@media(max-width: 1115px)": {
              height: "40vw"
            },
            "@media(max-width: 715px)": {
              height: "50vw"
            }
          }}
        >
          <iframe
            title="2018 QHacks Hackathon Highlights"
            src="https://www.youtube.com/embed/sqzy5n3m-RA?rel=0&amp;controls=0&amp;showinfo=0"
            frameBorder="0"
            allow="autoplay; encrypted-media"
            allowFullScreen
            css={{
              width: "500px",
              height: "281px",
              "@media(max-width: 1215px)": {
                width: "400px",
                height: "225px"
              },
              "@media(max-width: 1115px)": {
                paddingTop: "40px",
                width: "100%",
                height: "100%"
              }
            }}
          />
        </div>
        <picture
          css={{
            width: "275px",
            height: "196px",
            "@media(max-width: 1000px)": {
              width: "165px",
              height: "118px",
              bottom: "-70px"
            },
            position: "absolute",
            bottom: "-140px",
            right: "40px"
          }}
        >
          <source srcSet={cubes1webp} type="image/webp" />
          <source srcSet={cubes1png} type="image/png" />
          <img
            css={{
              width: "275px",
              height: "196px",
              "@media(max-width: 1000px)": {
                width: "165px",
                height: "118px"
              }
            }}
            src={cubes1png}
            alt="Floating cubes"
          />
        </picture>
      </div>
    </ContentWrapper>
  </div>
);

export default BriefSummary;
