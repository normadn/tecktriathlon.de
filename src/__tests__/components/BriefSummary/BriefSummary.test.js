import React from "react";
import BriefSummary from "../../../components/BriefSummary";
import { shallow } from "enzyme";

describe("BriefSummary", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <BriefSummary
          style={{ padding: "10px", margin: "10px" }}
          text="tEsT"
          lower={false}
          backgroundColor="#000000"
          foregroundColor="#ffffff"
        />
      )
    ).toMatchSnapshot();
  });
});
