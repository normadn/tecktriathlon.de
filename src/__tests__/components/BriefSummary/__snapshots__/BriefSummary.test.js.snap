// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BriefSummary renders correctly 1`] = `
<div
  css={
    Object {
      "background": "#f8f8f8",
      "paddingBottom": "80px",
      "paddingTop": "80px",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <_default>
    <div
      css={
        Object {
          "@media(max-width: 1115px)": Object {
            "display": "block",
          },
          "@media(max-width: 750px)": Object {
            "padding": "32px 24px 32px 24px",
          },
          "background": "#ffffff",
          "border": "solid 1px #e8e8e8",
          "borderRadius": "8px",
          "boxShadow": "4px 9px 40px 2px rgba(7, 16, 75, 0.1)",
          "display": "grid",
          "gridTemplateColumns": "auto auto",
          "margin": "0 auto 0 auto",
          "padding": "80px 55px 80px 55px",
        }
      }
    >
      <div
        css={
          Object {
            "@media (max-width: 920px)": Object {
              "paddingRight": "0px",
            },
            "paddingRight": "43px",
          }
        }
      >
        <h1
          css={
            Object {
              "paddingTop": "2px",
            }
          }
        >
          Dream It. Build It.
        </h1>
        <div
          css={
            Object {
              "display": "grid",
              "gridColumnGap": "4px",
              "gridTemplateColumns": "auto auto 1fr",
              "padding": "13px 0px 27px 0px",
            }
          }
        >
          <div
            data-css-g4rvbm=""
          />
          <div
            data-css-129fazg=""
          />
          <div
            data-css-1ectg3h=""
          />
        </div>
        <p
          css={
            Object {
              "fontColor": "#000000",
              "lineHeight": "24px",
            }
          }
        >
          QHacks is back in 2019 for another year of collaboration, ideation, and creativity! On February 1st - 3rd, more than 500 students from all over North America will come together to turn their ideas into reality. Attend workshops, play with hardware, take part in hacker challenges, and meet some new friends. Out-of-the-box thinkers, creatives, and innovators from all faculties and skill levels are encouraged to attend. We can't wait to see you there!
        </p>
      </div>
      <div
        css={
          Object {
            "@media(max-width: 1115px)": Object {
              "height": "40vw",
            },
            "@media(max-width: 715px)": Object {
              "height": "50vw",
            },
            "margin": "auto auto auto auto",
          }
        }
      >
        <iframe
          allow="autoplay; encrypted-media"
          allowFullScreen={true}
          css={
            Object {
              "@media(max-width: 1115px)": Object {
                "height": "100%",
                "paddingTop": "40px",
                "width": "100%",
              },
              "@media(max-width: 1215px)": Object {
                "height": "225px",
                "width": "400px",
              },
              "height": "281px",
              "width": "500px",
            }
          }
          frameBorder="0"
          src="https://www.youtube.com/embed/sqzy5n3m-RA?rel=0&controls=0&showinfo=0"
          title="2018 QHacks Hackathon Highlights"
        />
      </div>
      <picture
        css={
          Object {
            "@media(max-width: 1000px)": Object {
              "bottom": "-70px",
              "height": "118px",
              "width": "165px",
            },
            "bottom": "-140px",
            "height": "196px",
            "position": "absolute",
            "right": "40px",
            "width": "275px",
          }
        }
      >
        <source
          srcSet="test-file-stub"
          type="image/webp"
        />
        <source
          srcSet="test-file-stub"
          type="image/png"
        />
        <img
          alt="Floating cubes"
          css={
            Object {
              "@media(max-width: 1000px)": Object {
                "height": "118px",
                "width": "165px",
              },
              "height": "196px",
              "width": "275px",
            }
          }
          src="test-file-stub"
        />
      </picture>
    </div>
  </_default>
</div>
`;
